File: selfstudy/js/players_sum.js
/*
task 61:
Players - Sum
You need to calculate the total sum of player heights in cm
*/

function total_players(heights) {
  return heights.reduce((a, b) => a + b, 0);
}

// Tests
const tests = [
  { input: [10, 20, 30], expected: total_players([10, 20, 30]) },
  { input: [50, 60, 70], expected: total_players([50, 60, 70]) },
  { input: [1, 2, 3], expected: total_players([1, 2, 3]) },
  { input: [100, 200], expected: total_players([100, 200]) },
  { input: [], expected: total_players([]) }
];

if (process.argv.includes('mocha')) {
  const assert = require('assert');
  describe('total_players', () => {
    tests.forEach((t, idx) => {
      it(`test case ${idx+1}`, () => {
        assert.deepStrictEqual(total_players(t.input), t.expected);
      });
    });
  });
} else {
  tests.forEach((t, idx) => {
    console.log(`Test ${idx+1}:`, total_players(t.input) === t.expected || JSON.stringify(total_players(t.input)) === JSON.stringify(t.expected) ? "PASS" : "FAIL");
  });
}

// done

File: selfstudy/golang/players_sum.go
/*
task 261:
Players - Sum
You need to calculate the total sum of player heights in cm
*/

package main

import (
	"fmt"
	"reflect"
)

func TotalPlayers(heights []int) int {
	sum := 0
	for _, v := range heights {
		sum += v
	}
	return sum
}

func main() {
	tests := []struct {
		input []int
		expected int
	}{
		{[]int{10, 20, 30}, TotalPlayers([]int{10, 20, 30})},
		{[]int{50, 60, 70}, TotalPlayers([]int{50, 60, 70})},
		{[]int{1, 2, 3}, TotalPlayers([]int{1, 2, 3})},
		{[]int{100, 200}, TotalPlayers([]int{100, 200})},
		{[]int{}, TotalPlayers([]int{})},
	}
	
	for idx, t := range tests {
		res := TotalPlayers(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// done
